core/src/main/java/tc/oc/pgm/db/UsernameImpl.java

```diff
@@ -36,11 +36,10 @@ public String getName() {
   }
 
   @Override
-  public Component getStyledName(NameStyle style) {
+  public Component getName(NameStyle style) {
     return name == null
-        ? new PersonalizedText(
-            new PersonalizedTranslatable("misc.unknown"), ChatColor.DARK_AQUA, ChatColor.ITALIC)
-        : new PersonalizedPlayer(Bukkit.getPlayer(id), name, style);
+        ? TranslatableComponent.of("misc.unknown", TextColor.DARK_AQUA, TextDecoration.ITALIC)
+        : new PersonalizedPlayer(Bukkit.getPlayer(id), name, style).render();
```

Comment:
I think we should define a `PlayerComponent` interface, which contains static methods for building a player `TextComponent`.

So something like this:
```java
interface PlayerComponent {
  static TextComponent of(Player player);
  static TextComponent of(UUID playerId); // could show "Unknown"
  static TextComponent of(CommandSender sender);  // could show "Console" when null or console sender
}

Reply:
I created a `PlayerComponent` interface with the static methods you requested. Though I included `NameStyle` as a parameter, let me know if that’s how you envisioned it. If not any additional guidance would be great 👍